src/services/LinkService.ts

```diff
@@ -0,0 +1,107 @@
+import { LinkRepo } from "@src/repos";
+import { ApplicationError, HttpStatusCodes, LINKS } from "@src/common";
+import { generateShortId, prisma } from "@src/util";
+import { LinkServiceResult } from "@src/types";
+
+/**
+ * Add one link.
+ */
+async function addOne(url: string): LinkServiceResult {
+  try {
+    if (!url) {
+      return [
+        null,
+        new ApplicationError(LINKS.ERROR_MESSAGES.SERVICE_ERROR_LINKS, {
+          errorCode: LINKS.ERROR_CODES.URL_FOR_CONVERTING_NOT_PROVIDED,
+          statusCode: HttpStatusCodes.BAD_REQUEST,
+        }),
+      ];
```

Comment:
you can optimize this as i can see it's used accoss the app.
      
```suggestion
      return createResponse(null, {
     msg: LINKS.ERROR_MESSAGES.SERVICE_ERROR_LINKS, 
          errorCode: LINKS.ERROR_CODES.URL_FOR_CONVERTING_NOT_PROVIDED,
          statusCode: HttpStatusCodes.BAD_REQUEST,
        
      })
      
      
```

```
function createResponse(data?: whatever, error: { msg: string;
etc...
}) {

return [
        data,
        new ApplicationError(error.msg, {
          errorCode: error.errorCode,
          statusCode: error.errorCode,
        }),
      ];
      }
      
```

Reply:
implemented